compiler/llvm/clangUtil.cpp

```diff
@@ -2974,6 +2984,62 @@ void makeBinaryLLVM(void) {
   codegen_makefile(&mainfile, &tmpbinname, true);
   INT_ASSERT(tmpbinname);
 
+  if (fLibraryCompile) {
+    makeLLVMLibrary(moduleFilename, tmpbinname, dotOFiles);
+  } else {
+    // Runs the LLVM link command
+    runLLVMLinking(useLinkCXX, options, moduleFilename, maino, tmpbinname,
+                   dotOFiles, clangLDArgs, saw_sysroot);
+  }
+
+
+  // If we're not using a launcher, copy the program here
+  if (0 == strcmp(CHPL_LAUNCHER, "none")) {
+
+    if (fLibraryCompile) {
+      moveGeneratedLibraryFile(tmpbinname);
+    } else {
+      moveGeneratedExecutable(tmpbinname);
+    }
+
+  } else {
+    // Now run the makefile to move from tmpbinname to the proper program
+    // name and to build a launcher (if necessary).
+    const char* makeflags = printSystemCommands ? "-f " : "-s -f ";
+    const char* makecmd = astr(astr(CHPL_MAKE, " "),
+                               makeflags,
+                               getIntermediateDirName(), "/Makefile");
+
+    if( printSystemCommands ) {
+      printf("%s\n", makecmd);
+      fflush(stdout); fflush(stderr);
+    }
+
+    mysystem(makecmd, "Make Binary - Building Launcher and Copying");
+  }
+}
+
+static void makeLLVMLibrary(std::string moduleFilename, const char* tmpbinname,
+                            std::vector<std::string> dotOFiles) {
+  std::string commandBase = "ar -c -r -s"; // Stolen from Makefile.static
```

Comment:
Usually it's possible to override the AR command in a Makefile setting. It might be worth offering a way to do to that (but that can be future work).

Reply:
Yeah - I saw that $(AR) was a Makefile variable but wasn't sure how to replicate that outside of a Makefile so hard-coded it for the moment.

Reply:
I opened #12505 to track that